### PR TITLE
fix Issue 14565 - dmd -profile produces garbled output for long-runni…

### PR DESCRIPTION
**Pull request by WalterBright**
…ng CPU-intensive processes

https://issues.dlang.org/show_bug.cgi?id=14565

Increased the counter from 32 bits to 64 bits.

### CODE REVIEW

File: src/rt/trace.d [hunk 1]
```diff
@@ -35,7 +35,7 @@ struct SymPair
 {
     SymPair* next;
     Symbol* sym;        // function that is called
-    uint count;         // number of times sym is called
+    long count;         // number of times sym is called
```
Comment (yebblies) [inline src/rt/trace.d]: Should be `ulong`?

Comment (WalterBright) [inline src/rt/trace.d]: yes. Oops!

File: src/rt/trace.d [hunk 2]
```diff
@@ -163,18 +163,23 @@ private int sympair_cmp(in void* e1, in void* e2)
     auto psp1 = cast(SymPair**)e1;
     auto psp2 = cast(SymPair**)e2;
 
-    return (*psp2).count - (*psp1).count;
+    long cmp = (*psp2).count - (*psp1).count;
```
Comment (yebblies) [inline src/rt/trace.d]: Doesn't this pattern produce the wrong results in some cases?  Isn't it better to just use `if (a < b) ...` and let the compiler optimize it?

**Comment by MartinNowak** [comment]
Auto-merge toggled on

**Pull request merged**